src/ray/util/pipe_logger.cc

```diff
@@ -216,57 +150,22 @@ bool ShouldUsePipeStream(const StreamRedirectionOption &stream_redirect_opt) {
          stream_redirect_opt.tee_to_stderr;
 }
 
-#if defined(__APPLE__) || defined(__linux__)
 RedirectionFileHandle OpenFileForRedirection(const std::string &file_path) {
-  int fd = open(file_path.data(), O_WRONLY | O_CREAT, 0644);
-  RAY_CHECK_NE(fd, -1) << "Fails to open file " << file_path << " with failure reason "
-                       << strerror(errno);
-
-  auto flush_fn = [fd]() {
-    RAY_CHECK_EQ(fsync(fd), 0) << "Fails to flush data to disk because "
-                               << strerror(errno);
+  boost::iostreams::file_descriptor_sink sink{file_path, std::ios_base::out};
+  auto handle = sink.handle();
+  auto ostream =
+      std::make_shared<boost::iostreams::stream<boost::iostreams::file_descriptor_sink>>(
+          std::move(sink));
+  auto flush_fn = [ostream]() { ostream->flush(); };
```

Comment:
make sure `fsync(fd)` is actually called underneath.

Reply:
That's really good point, I think we need to manually flush the fd.